discord_google_calendar_bot.py

```diff
@@ -0,0 +1,338 @@
+    event in the calendar, using the event ID from the old Discord event.
+    Finally, it prints a message to the console indicating that the Google
+    Calendar event has been updated.
+    """
+    google_event_summary_prefix = "Discord (" + discord_client.guilds[0].name + "): "
+
+    if new_discord_event.entity_type is discord.EntityType.voice:
+        discord_end_time = new_discord_event.start_time
+        discord_location = (
+            GOOGLE_EVENT_CHANNEL_NAME_PREFIX + new_discord_event.channel.name
+        )
+    else:
+        discord_end_time = new_discord_event.end_time
+        discord_location = GOOGLE_EVENT_LOCATION_PREFIX + new_discord_event.location
+
+    google_event_details = {
+        "id": str(new_discord_event.id),
+        "summary": google_event_summary_prefix + new_discord_event.name,
+        "location": discord_location,
+        "description": new_discord_event.description,
+        "start": {
+            "dateTime": new_discord_event.start_time.isoformat(),
+            "timeZone": "UTC",
+        },
+        "end": {
+            "dateTime": discord_end_time.isoformat(),
+            "timeZone": "UTC",
+        },
+    }
+
+    # Update the calendar event based on Discord event ID and announce it to
+    # the console
+    google_client.events().update(
+        calendarId=CALENDAR_ID,
+        eventId=str(old_discord_event.id),
+        body=google_event_details,
+    ).execute()
+    print(f"Google Calendar event updated for {new_discord_event.name}")
+
+
+@discord_client.event
+async def on_ready():
+    """
+    Event handler that is called when the Discord client is ready and logged
+    in.
+
+    This function performs the following tasks:
+
+    1. Prints a message to the console indicating that the Discord client
+    has logged in.
+    2. Generates a prefix for the Google Calendar event summaries,
+    which includes the name of the Discord server.
+    3. Retrieves all scheduled events from the first Discord server the
+    client is connected to.
+    4. Retrieves the current events from the Google Calendar.
+    5. Compares the scheduled Discord events to the existing Google Calendar
+    events, and creates new Google Calendar events for any Discord events
+    that don't have a matching Google Calendar event.
+
+    This function is automatically called by the Discord client when the
+    connection is established and the client is ready to receive and send
+    messages.
+    """
+    # Announce ourselves to the console
+    print(f"We have logged in as {discord_client.user}")
+
+    google_event_summary_prefix = "Discord (" + discord_client.guilds[0].name + "): "
+
+    # Do an initial push of all Discord scheduled events to Google Calendar,
+    # matching based on event summary field
+    discord_guild = discord_client.guilds[0]
+    discord_scheduled_events = discord_guild.scheduled_events
+    now = datetime.utcnow().isoformat() + "Z"  # 'Z' indicates UTC time
```

Comment:
utcnow is depercated

Reply:
This now uses `datetime.now(timezone.utc).replace(tzinfo=None).isoformat() + 'Z'` instead, which is not deprecated and exactly matches the format Google Calendar API expects